### PR TITLE
chore: removed travis yml and added git action support

### PR DESCRIPTION
**Pull request by ozayr-zaviar**
## Summary
- We remove travis yaml from ruby-sdk and moved all the stages to git action.

NOTE:
Please update [this](https://github.com/optimizely/ruby-sdk/blob/21005c3085d670e6b3b9c2993ac194163c146c0d/.github/workflows/ruby.yml#L11), [this](https://github.com/optimizely/ruby-sdk/blob/21005c3085d670e6b3b9c2993ac194163c146c0d/.github/workflows/ruby.yml#L14) and [this](https://github.com/optimizely/ruby-sdk/blob/21005c3085d670e6b3b9c2993ac194163c146c0d/.github/workflows/ruby.yml#L20) line branch name to master before merging.

## Test plan
All tests should pass

### CODE REVIEW

**Review by jaeopt — APPROVED** [review]
LGTM

**Review by msohailhussain — CHANGES_REQUESTED** [review]
please address.

**Review by msohailhussain — CHANGES_REQUESTED** [review]
please address.

**Review by jaeopt — APPROVED** [review]
LGTM

File: optimizely-sdk.gemspec [hunk 1]
```diff
@@ -17,7 +17,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
 
   spec.add_development_dependency 'bundler'
-  spec.add_development_dependency 'coveralls'
+  spec.add_development_dependency 'coveralls_reborn'
```
[the remark this answers is not in the document]
Comment (ozayr-zaviar) [reply optimizely-sdk.gemspec:20]: Coveralls is last updated on Sep 9, 2021 and a newer and updated fork is coveralls_reborn. Coveralls is also not supported with git actions.

**Review by msohailhussain — APPROVED** [review]
lgtm

File: .github/workflows/ruby.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  lint_markdown_files:
+    uses: optimizely/ruby-sdk/.github/workflows/lint_markdown.yml@master
+
+  integration_tests:
+    uses: optimizely/ruby-sdk/.github/workflows/integration_test.yml@uzair/master
```
Comment (msohailhussain) [inline .github/workflows/ruby.yml:14]:
```suggestion
    uses: optimizely/ruby-sdk/.github/workflows/integration_test.yml@master
```

File: .github/workflows/ruby.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    secrets:
+      CI_USER_TOKEN: ${{ secrets.CI_USER_TOKEN }}
+      TRAVIS_COM_TOKEN: ${{ secrets.TRAVIS_COM_TOKEN }}
+
+  fullstack_production_suite:
+    uses: optimizely/ruby-sdk/.github/workflows/integration_test.yml@uzair/master
```
Comment (msohailhussain) [inline .github/workflows/ruby.yml:20]:
```suggestion
    uses: optimizely/ruby-sdk/.github/workflows/integration_test.yml@master
```

**Pull request merged**